packages/cyberstorm/src/components/Forms/CreateTeamForm/CreateTeamForm.tsx

```diff
@@ -0,0 +1,118 @@
+"use client";
+import styles from "./CreateTeamForm.module.css";
+import * as Button from "../../Button";
+import { TextInput } from "../../TextInput/TextInput";
+import { useForm, FieldErrors, useController } from "react-hook-form";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faArrowsRotate } from "@fortawesome/free-solid-svg-icons";
+
+import { useContext } from "react";
+import { ToastContext } from "../../Toast/ToastContext";
+import {
+  faCircleCheck,
+  faCircleExclamation,
+  faOctagonExclamation,
+} from "@fortawesome/pro-solid-svg-icons";
+
+/**
+ * Form for creating a team
+ */
+export function CreateTeamForm() {
+  const useToast = () => useContext(ToastContext);
+  const toast = useToast();
+
+  async function onValid(data: { teamName: string }) {
+    // TODO: Add sending to TS API endpoint
+    toast?.addToast(
+      "info",
+      <FontAwesomeIcon icon={faCircleExclamation} />,
+      "Creating team"
+    );
+    await new Promise((resolve) => {
+      setTimeout(resolve, 1000);
+    });
+    // TODO: Add a toast on response return based on the response
+    toast?.addToast(
+      "success",
+      <FontAwesomeIcon icon={faCircleCheck} />,
+      "Team created"
+    );
+    console.log(JSON.stringify(data));
+  }
+
+  async function onInvalid(
```

Comment:
I believe this is exactly the boilerplatey approach Mythic asked to avoid. Did you check the Zod option? Did you try to generalize this approach to a reusable helper?

Reply:
I think I've reduced boilerplate a lot in the fix comment. Used zod